server/user.js

```diff
@@ -0,0 +1,85 @@
+const { TimeLogger } = require("../src/util");
+const { R } = require("redbean-node");
+const { UptimeKumaServer } = require("./uptime-kuma-server");
+const server = UptimeKumaServer.getInstance();
+const io = server.io;
+
+/**
+ * Send list of users to client
+ * @param {Socket} socket Socket.io socket instance
+ * @returns {Promise<Bean[]>}
+ */
+async function sendUserList(socket) {
+    const timeLogger = new TimeLogger();
+
+    let result = [];
+    let list = (await R.findAll("user")).map(({ id, username, active }) => ({
+        id,
+        username,
+        active
+    }));
+
+    io.to(socket.userID).emit("userList", result);
+
+    timeLogger.print("Send User List");
+
+    return list;
+}
+
+/**
+ * Send list of users to client
+ * @param {Socket} socket Socket.io socket instance
+ * @returns {Promise<Bean[]>}
+ */
+async function getUser(userID) {
+    const timeLogger = new TimeLogger();
+
+    const bean = await R.findOne("user", " id = ? ", [ userID ])
+        .then(({ id, username, active }) => ({
+            id,
+            username,
+            active
+        }));
+
+    if (!bean) {
+        throw new Error("User not found");
+    }
+
+    timeLogger.print(`Get user ${userID}`);
+
+    return bean;
+}
+
+/**
+ * Saves and updates given user entity
+ * @param {Socket} socket Socket.io socket instance
+ * @param user
+ * @returns {Promise<Bean[]>}
+ */
+async function saveUser(socket, user) {
+    const timeLogger = new TimeLogger();
+    const { id, username, active } = user;
+
+    const bean = await R.findOne("user", " id = ? ", [ id ]);
+
+    if (!bean) {
+        throw new Error("User not found");
+    }
+
+    username && (bean.username = username);
+    active !== undefined && (bean.active = active);
+
+    await R.store(bean);
+
+    io.to(socket.userID).emit("saveUser", bean);
+
+    timeLogger.print(`Save user ${user.id}`);
+
+    return bean;
```

Comment:
Why have you added this return statement? (it is not being used where the function is called from)

```suggestion
```

Reply:
See above: https://github.com/M1CK431/uptime-kuma/pull/1#discussion_r1287713970